src/AlgAssAbsOrd/Ideal.jl

```diff
@@ -1637,7 +1637,7 @@ function _as_order_of_smaller_algebra(m::AbsAlgAssMor, O::AlgAssAbsOrd, OB::AlgA
   basis_in_A = Vector{elem_type(A)}(undef, dim(B))
   for i = 1:dim(B)
     t = elem_from_mat_row(OB, N, i)
-    b, s = haspreimage(m, elem_in_algebra(t, copy = false))
+    b, s = has_preimage_with_preimage(m, elem_in_algebra(t, copy = false))
```

Comment:
In a usage pattern like this, would it be better to just call `preimage`? the boolean here is only used to throw an error, so calling preimage directly should do the job?

Reply:
Probably yes, I did not check all changes at once but if there is a `preimage` function defined, it should do the job. Eventually, I would expect the `preimage` function to do exactly the same: call `has_preimage_with_preimage` and return an error if the boolean is `false`. I could add a `preimage` function when missing and change those calls accordingly if you think it is better.